packages/react-centra-checkout/src/Context/index.tsx

```diff
@@ -232,34 +232,35 @@ export function CentraProvider(props: ProviderProps) {
   }
 
   const selectionApiCall = useCallback(
-    async (
-      apiCall:
-        | Promise<CheckoutApi.Response<CheckoutApi.SelectionResponse>>
-        | (() => Promise<CheckoutApi.Response<CheckoutApi.SelectionResponse>>),
-    ) => {
+    async (apiCall: Promise<unknown> | (() => Promise<unknown>)) => {
       window.CentraCheckout?.suspend()
       const response = typeof apiCall === 'function' ? await apiCall() : await apiCall
 
-      if ('selection' in response && response.selection) {
-        setSelection(response)
+      if (
+        Boolean(response) &&
+        response !== null &&
+        typeof response === 'object' &&
+        'selection' in response &&
+        response.selection
+      ) {
+        setSelection(response as CheckoutApi.SuccessResponse<CheckoutApi.SelectionResponse>)
```

Comment:
In result to `ApiClient.request` returning `Promise<unknown>`, we had to perform some additional type narrowing. This wasn't enforced by TS, since the the returned value was `Promise<any>`

Reply:
question: I understand the "if statement" refactor based off of the new type given to the `apiCall` argument. However, I'm not sure I follow as to why the type for `apiCall` was refactored to use `unknown` instead of `CheckoutApi.Response<CheckoutApi.SelectionResponse>`. Could you elaborate?

Reply:
The main reason is that we otherwise would have to do the "if statement" in all places where we call `selectionApiCall`.
We could though, move the "if statement" to a separate helper function, wrap the parameter passed to `selectionApiCall` in every call. By doing so, we could revert the changes made in `selectionApiCall`, @maeertin

Reply:
Chiming in here! I think we should do the if statement in a helper function like you said @adamsoderstrom! Because the `selectionApiCall` should actually expect an apiCall that returns a SelectionResponse, otherwise you could pass whatever into it, and then it doesn't make sense to call it a `selectionApiCall`

Reply:
I've now done what you've mentioned above, @alexanderflink.
Though, i wonder how we want this design change to be exposed from the handlers as well?
For example, returning `Promise<unknown>` instead of `Promise<CheckoutApi.SuccessResponse<CheckoutApi.Selection>>`?

Reply:
Consider my earlier comment out of scope for this PR.
This PR will not obfuscate the exposed return types of the handlers in `Context`.